### PR TITLE
migrating to melodic

### PR DESCRIPTION
**Pull request by ahcorde**

### CODE REVIEW

File: moveit_cartesian_plan_plugin/CMakeLists.txt [hunk 1]
```diff
@@ -11,21 +13,54 @@ find_package(catkin REQUIRED COMPONENTS
...
+else()
+  find_package(Qt5Widgets REQUIRED)
+  find_package(Qt5Concurrent REQUIRED)
+	set(QT_LIBRARIES Qt5::Widgets Qt5::Concurrent)
+endif()
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+set(CMAKE_AUTOMOC ON)
 add_definitions(-DQT_NO_KEYWORDS)
+add_definitions(-std=c++11)
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/CMakeLists.txt]: This is redundant as there already is a -std=c++11 compile option at the top of the file

File: moveit_cartesian_plan_plugin/CMakeLists.txt [hunk 2]
```diff
@@ -11,21 +13,54 @@ find_package(catkin REQUIRED COMPONENTS
   rqt_gui_cpp
   rviz
   visualization_msgs
-  moveit_ros_move_group
-  moveit_ros_planning_interface
   tf
   tf_conversions
+  moveit_ros_move_group
+  moveit_ros_planning_interface
+  moveit_ros_warehouse
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/CMakeLists.txt]: Are the warehouse or perception libraries necessary?

File: moveit_cartesian_plan_plugin/CMakeLists.txt [hunk 3]
```diff
@@ -11,21 +13,54 @@ find_package(catkin REQUIRED COMPONENTS
...
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+set(CMAKE_AUTOMOC ON)
 add_definitions(-DQT_NO_KEYWORDS)
+add_definitions(-std=c++11) 
+
+# # include(${QT_USE_FILE})
+#
+# ## I prefer the Qt signals and slots to avoid defining "emit", "slots",
+# ## etc because they can conflict with boost signals, so define QT_NO_KEYWORDS here.
+# add_definitions(-DQT_NO_KEYWORDS)
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/CMakeLists.txt]: Remove commented commands

File: moveit_cartesian_plan_plugin/CMakeLists.txt [hunk 4]
```diff
@@ -64,12 +104,39 @@ catkin_package(
   visualization_msgs
 )
 
-qt4_wrap_cpp(moveit_cartesian_plan_plugin_MOCS ${moveit_cartesian_plan_plugin_HDRS})
-qt4_wrap_ui(moveit_cartesian_plan_plugin_UIS_H ${moveit_cartesian_plan_plugin_UIS})
+# qt4_wrap_cpp(moveit_cartesian_plan_plugin_MOCS ${moveit_cartesian_plan_plugin_HDRS})
+# qt4_wrap_ui(moveit_cartesian_plan_plugin_UIS_H ${moveit_cartesian_plan_plugin_UIS})
+
+# qt_wrap_ui(moveit_cartesian_plan_plugin_UIS_H ${moveit_cartesian_plan_plugin_UIS})
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/CMakeLists.txt]: remove commented commands

File: moveit_cartesian_plan_plugin/CMakeLists.txt [hunk 5]
```diff
@@ -64,12 +104,39 @@ catkin_package(
...
+
+# ensure generated header files are being created in the devel space
+set(_cmake_current_binary_dir "${CMAKE_CURRENT_BINARY_DIR}")
+set(CMAKE_CURRENT_BINARY_DIR "${CATKIN_DEVEL_PREFIX}/${CATKIN_GLOBAL_INCLUDE_DESTINATION}")
+
+if(rviz_QT_VERSION VERSION_LESS "5")
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/CMakeLists.txt]: This qt library resolution is already done above

File: moveit_cartesian_plan_plugin/CMakeLists.txt [hunk 6]
```diff
@@ -64,12 +104,39 @@ catkin_package(
...
 add_library(${PROJECT_NAME} ${moveit_cartesian_plan_plugin_SRCS} ${moveit_cartesian_plan_plugin_MOCS} ${moveit_cartesian_plan_plugin_UIS_H}) # ${moveit_cartesian_plan_plugin_UIS_H}
-target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${moveit_move_group_interface} yaml-cpp)
+target_link_libraries(${PROJECT_NAME}
+  ${catkin_LIBRARIES}
+  ${moveit_move_group_interface}
+  yaml-cpp
+  ${Boost_LIBRARIES}
+  ${QT_LIBRARIES})
+  # ${QT_QTCORE_LIBRARY}
+  # ${QT_QTGUI_LIBRARY}
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/CMakeLists.txt]: Remove commented commands

File: moveit_cartesian_plan_plugin/include/moveit_cartesian_plan_plugin/add_way_point.hpp [hunk 7]
```diff
@@ -27,17 +27,18 @@
...
 
 #include <QWidget>
 #include <QCursor>
 #include <QObject>
 #include <QKeyEvent>
 #include <QHBoxLayout>
 #include <QTimer>
-#include <QtConcurrentRun>
+#include <QtConcurrent/QtConcurrent>
+// #include <QtConcurrentRun>
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/include/moveit_cartesian_plan_plugin/add_way_point.hpp]: Remove commented code

File: moveit_cartesian_plan_plugin/include/moveit_cartesian_plan_plugin/generate_cartesian_path.hpp [hunk 8]
```diff
@@ -1,7 +1,17 @@
...
+
+//#include <moveit/planning_interface/move_group_interface.h>
+
+#include <moveit/move_group_interface/move_group_interface.h>
+
+// #include <moveit/planning_interface/move_group_interface.h>
+// #include <moveit/move_group_interface/move_group_interface.h>
 #include <moveit/robot_trajectory/robot_trajectory.h>
 #include <moveit/robot_model_loader/robot_model_loader.h>
+// #include <moveit/planning_interface/planning_interface.h>
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/include/moveit_cartesian_plan_plugin/generate_cartesian_path.hpp]: Same as above

File: moveit_cartesian_plan_plugin/include/moveit_cartesian_plan_plugin/generate_cartesian_path.hpp [hunk 9]
```diff
@@ -12,14 +22,17 @@
 
 #include <QObject>
 #include <QTimer>
-#include <QtConcurrentRun>
+// #include <QtConcurrentRun>
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/include/moveit_cartesian_plan_plugin/generate_cartesian_path.hpp]: Same as above

File: moveit_cartesian_plan_plugin/include/moveit_cartesian_plan_plugin/generate_cartesian_path.hpp [hunk 10]
```diff
@@ -1,7 +1,17 @@
 // MoveIt!
-#include <moveit/move_group_interface/move_group.h>
+//#include <moveit/move_group_interface/move_group.h>
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/include/moveit_cartesian_plan_plugin/generate_cartesian_path.hpp]: Why change the extension of the file to **hpp**?

File: moveit_cartesian_plan_plugin/include/moveit_cartesian_plan_plugin/generate_cartesian_path.hpp [hunk 11]
```diff
@@ -12,14 +22,17 @@
...
+// #include <QtConcurrentRun>
+#include <QtConcurrent/QtConcurrent>
 #include <QFuture>
 
 
 #ifndef GENERATE_CARTESIAN_PATH_H_
 #define GENERATE_CARTESIAN_PATH_H_
 
-typedef boost::shared_ptr<move_group_interface::MoveGroup> MoveGroupPtr;
+namespace move_group_interface = moveit::planning_interface;
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/include/moveit_cartesian_plan_plugin/generate_cartesian_path.hpp:24]: this namespace isn't being used

File: moveit_cartesian_plan_plugin/include/moveit_cartesian_plan_plugin/widgets/path_planning_widget.hpp [hunk 12]
```diff
@@ -13,11 +13,12 @@
 
 #include <ui_path_planning_widget.h>
 
-#include <moveit_cartesian_plan_plugin/add_way_point.h>
+#include <moveit_cartesian_plan_plugin/add_way_point.hpp>
 
 #include <QWidget>
 #include <QTimer>
-#include <QtConcurrentRun>
+// #include <QtConcurrentRun>
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/include/moveit_cartesian_plan_plugin/widgets/path_planning_widget.hpp]: Remove commented code

File: moveit_cartesian_plan_plugin/plugin.xml [hunk 13]
```diff
@@ -6,12 +6,5 @@
       Tool for adding spherical way points for trajectory generation.
     </description>
   </class>
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/plugin.xml:8]: Did this plugin get deleted?

File: moveit_cartesian_plan_plugin/src/generate_cartesian_path.cpp [hunk 14]
```diff
@@ -4,7 +4,7 @@
 #include <math.h>
 #include <moveit/robot_state/conversions.h>
 #include <moveit/trajectory_processing/iterative_time_parameterization.h>
-#include <moveit_cartesian_plan_plugin/generate_cartesian_path.h>
+#include <moveit_cartesian_plan_plugin/generate_cartesian_path.hpp>
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/src/generate_cartesian_path.cpp:7]: What functionality does this header bring in that wasn't here before?

File: moveit_cartesian_plan_plugin/src/point_tree_model.cpp [hunk 15]
```diff
@@ -1,7 +1,7 @@
-#include <QtGui>
+#include <QtWidgets>
 
-#include <moveit_cartesian_plan_plugin/point_tree_item.h>
-#include <moveit_cartesian_plan_plugin/point_tree_model.h>
+#include <moveit_cartesian_plan_plugin/point_tree_item.hpp>
+#include <moveit_cartesian_plan_plugin/point_tree_model.hpp>
```
Comment (jrgnicho) [inline moveit_cartesian_plan_plugin/src/point_tree_model.cpp:4]: Why change the extension to **hpp**?

**Review by jrgnicho — CHANGES_REQUESTED** [review]
@ahcorde  thanks for submitting this.  Could you please address the comments;  I'll test this on a melodic system soon.

**Comment by ahcorde** [comment]
reviewed it

**Comment by rkojcev** [comment]
addressed some of the @jrgnicho comments and opened a new PR in https://github.com/ros-industrial-consortium/fermi/pull/50. Thanks @ahcorde for helping.

**Pull request closed without merging**